doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+	  * [3.4 Syncd](#34-syncd)
+	  * [3.5 SAI](#35-sai)
+	  * [3.6 CLI](#36-cli)
+		  * [3.6.1 Configuration Commands](#361-configuration-commands)
+		  * [3.6.2 Show Commands](#362-show-commands)
+  * [4. Flow Diagrams](#4-flow-diagrams)
+  * [5. Serviceability and Debug](#5-serviceability-and-debug)
+  * [6. Warm Boot Support](#6-warm-boot-support)
+  * [7. Scalability](#7-scalability)
+  * [8. Unit Test](#8-unit-test)
+	
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                         |
+|:---:|:-----------:|:------------------:|--------------------------------------------|
+| 1.0 | 04/30/2019  |   Anil Pandey      | Added detailed requirements                |
+|     |             |                    | Added detailed Unit Test cases             |
+| 0.3 | 04/29/2019  |   Pankaj Jain      | Modified static-mac and aging commands     |
+| 0.2 | 04/26/2019  |   Anil Pandey      | Merged VLAN Range FS contents              |
+| 0.1 | 04/11/2019  |   Anil Pandey      | Initial version                            |
+
+
+# About this Manual
+This document provides general information about the Layer 2 Forwarding Enhancements feature implementation in SONiC.
+# Scope
+This document describes the high level design of Layer 2 Forwarding Enhancements feature. 
+
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+|   FDB                    |  Forwarding Database                |
+
+
+# 1 Requirement Overview
+## 1.1 Functional Requirements
+
+ 1. FDB Flush Support
+	 - FDB entries should be flushed per Port when the operational state of the port goes down. Only dynamic entries should be flushed.
+	 - FDB entries should be flushed per Port per VLAN when a port is removed from a VLAN. Both dynamic and static entries should be flushed. Static FDB configuration should not be removed.
+	 - FDB entries should be flushed per Port or per Port per VLAN when triggered by Layer 2 Protocol upon topology change. Only dynamic entries should be flushed.
+	 - FDB entries should be flushed per Portchannel when the admin or operational status of the Portchannel goes down.
+	 - FDB entry should be removed from FDB_TABLE in STATE_DB and ASIC_SB and internal orchagent FDB data structure and Hardware.
+	
+2. Handle MAC move event generated by hardware.
+	- Existing FDB entry should be replaced with one having new Port in ASIC_DB and STATE_DB when a MAC move event is received.
+	
+3. Configuration CLI for FDB aging time.
+	- The FDB aging time should be configurable in hardware to a desired value from CLI.
+	- There should be an option to disable aging by setting the aging time to 0.
+	- The default FDB aging time should be set to 0. User has option to set desired value in default config file.
```

Comment:
Default to 0 is weird. How about other vendor OS or standard default value?

Reply:
@qiluo-msft , as per SAI definition default value is 0 which treats AGING as disabled. Looks ok to me. 

```
     * Zero means aging is disabled.
     *
     * @type sai_uint32_t
     * @flags CREATE_AND_SET
     * @default 0
     */
    SAI_SWITCH_ATTR_FDB_AGING_TIME,
```

Reply:
Discussed with Qi, and I agree that for Sonic, the default is not disabled but setting to value as 600sec. Azure/sonic-buildimage#2365. Please update!

Reply:
Hi Prince, the idea is to have a config file for each profile (L2, L3) and set the default aging time in the config. The default aging time 0 mentioned in the FS is for the aging time setting when SONiC is initialized. Let me know what you think.

Reply:
Updated doc with default aging time as 600 seconds.